WordPress/gutenberg
# Components: Add accessible Toolbar

## Description


This PR is a simpler version of #17875. It adds an `__experimentalAccessibilityLabel` prop to the `Toolbar` component in `@wordpress/components`. If this prop is passed in, then it will render an accessible toolbar. Otherwise, it'll fallback to the current toolbar on master (which is not actually a toolbar) that has been renamed to `ToolbarGroup`.

This `Toolbar` only accepts `ToolbarButton` as toolbar items. `ToolbarButton` is currently an enhanced `IconButton`, but it should be improved in a follow-up PR.

With this, projects using `@wordpress/components` could start experimenting it for simple toolbars.

### Story

1. This PR

2. Improve `ToolbarButton` so that it serves more use cases than just being an `IconButton`.

    At this point, the Gutenberg header toolbar (without the fixed block toolbar) could use it.

3. Refactor `SlotFill` so we won't need to force fills to re-render when `fillProps` change.

    At this point, block toolbars could start using it.

4. Convert the fixed toolbar buttons (block switcher, more tools etc.) to `ToolbarButton`.

    At this point, custom blocks could start using it, as long as they use `ToolbarButton` inside `BlockControls` or `BlockFormatControls`.

5. Convert core block toolbars.

    We could convert one per PR (for example, start with the Paragraph toolbar).

6. Let it be for a while.

7. If it proves to be useful and works out, deprecate the old usage and remove the `__experimental` prefix. Otherwise, revert it altogether, which would break the code for people using `__experimentalAccessibilityLabel` explicitly, but custom blocks would seamlessly rollback to the old usage.

Related: #15331 #3383 #16514 #17875

## How has this been tested?



1. Run `npm run storybook:dev`
2. See the different `Toolbar` stories

## Screenshots 

![Oct-26-2019 22-52-06](https://user-images.githubusercontent.com/3068563/67629920-e6c9e880-f843-11e9-8332-f96bab3bc5f1.gif)

## Types of changes





New feature

## Checklist:
